### PR TITLE
Removed curl brackets from FROM queries ( ) for MariaDB 10.4 support.

### PR DESCRIPTION
**Pull request by tonypartridge**

### CODE REVIEW

File: component/site/views/flat/icals/tmpl/ical_form.php [hunk 1]
```diff
@@ -27,7 +27,7 @@
 			continue;
 		$cats[] = $cid;
 	}
-	if (count($cats) == 0)
+	if (!empty($cats) && count($cats) == 0)
```
Comment (GeraintEdwards) [inline component/site/views/flat/icals/tmpl/ical_form.php]: See https://www.php.net/manual/en/function.empty.php if $cats is array() then empty returns true.  So your logic cancels out here.

Is the purpose of this change to allow export of all categories?

File: component/site/views/flat/icals/tmpl/ical_form.php [hunk 2]
```diff
@@ -37,7 +37,7 @@
 	$cats = implode(",", $cats);
 
 	$link = JURI::root() . "index.php?option=com_jevents&task=icals.export&format=ical";
-	if (count($cats) > 0)
+	if (!empty($cats) && count($cats) > 0)
```
Comment (GeraintEdwards) [inline component/site/views/flat/icals/tmpl/ical_form.php]: see above - this is testing the same thing twice

File: component/admin/controllers/icalrepeat.php [hunk 3]
```diff
@@ -171,7 +171,7 @@ function edit($key = NULL, $urlVar = NULL)
 
 		$db = JFactory::getDbo();
 		$query = "SELECT rpt.eventid"
-				. "\n FROM (#__jevents_vevent as ev, #__jevents_icsfile as icsf)"
+				. "\n FROM #__jevents_vevent as ev, #__jevents_icsfile as icsf"
```
Comment (GeraintEdwards) [inline component/admin/controllers/icalrepeat.php]: I'm presuming this still works in MySQL ?

Should probably be an INNER JOIN

Comment (tonypartridge) [reply component/admin/controllers/icalrepeat.php]: Maybe, but I'm just improving mariadb support :P

File: component/site/views/flat/icals/tmpl/ical_form.php (hunk 2 above)
Comment (tonypartridge) [reply component/site/views/flat/icals/tmpl/ical_form.php]: set to is_Array

File: component/site/views/flat/icals/tmpl/ical_form.php (hunk 1 above)
Comment (tonypartridge) [reply component/site/views/flat/icals/tmpl/ical_form.php]: set to is_array

**Pull request merged**